presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -105,10 +108,85 @@ public void testTableHandleCached()
+
+        createTable(phantomTable);
+        JdbcTableHandle tableHandle = jdbcClient.getTableHandle(SESSION, phantomTable).orElseThrow();
+
+        assertThatThrownBy(() -> cachingJdbcClient.getColumns(SESSION, tableHandle))
+                .isInstanceOf(TableNotFoundException.class);
+
+        JdbcColumnHandle phantomColumn = addColumn(tableHandle);
+
+        assertThat(cachingJdbcClient.getColumns(SESSION, tableHandle))
+                .containsExactly(phantomColumn);
+
+        cachingJdbcClient.dropTable(SESSION, tableHandle);
+
+        assertThatThrownBy(() -> cachingJdbcClient.getColumns(SESSION, tableHandle))
+                .isInstanceOf(TableNotFoundException.class);
+    }
+
+    @Test
+    public void shouldInvalidateTableCachesOnColumnDrop()
+    {
+        SchemaTableName phantomTable = new SchemaTableName(schema, randomTableName());
+        assertThat(cachingJdbcClient.getTableHandle(SESSION, phantomTable)).isEmpty();
+
+        createTable(phantomTable);
+        JdbcTableHandle tableHandle = jdbcClient.getTableHandle(SESSION, phantomTable).orElseThrow();
+
+        assertThatThrownBy(() -> cachingJdbcClient.getColumns(SESSION, tableHandle))
+                .isInstanceOf(TableNotFoundException.class);
+
+        JdbcColumnHandle phantomColumn = addColumn(tableHandle);
+        assertThat(cachingJdbcClient.getColumns(SESSION, tableHandle))
+                .containsExactly(phantomColumn);
+
+        JdbcColumnHandle otherColumn = addColumn(cachingJdbcClient, tableHandle, "other_column");
+        assertThat(cachingJdbcClient.getColumns(SESSION, tableHandle))
+                .containsExactly(phantomColumn, otherColumn);
+
+        cachingJdbcClient.dropColumn(SESSION, tableHandle, phantomColumn);
+
+        assertThat(cachingJdbcClient.getColumns(SESSION, tableHandle))
+                .containsExactly(otherColumn);
+
+        cachingJdbcClient.dropTable(SESSION, tableHandle);
+    }
+
+    @Test
+    public void shouldInvalidateTableCachesOnColumnRename()
+    {
+        SchemaTableName phantomTable = new SchemaTableName(schema, randomTableName());
+        assertThat(cachingJdbcClient.getTableHandle(SESSION, phantomTable)).isEmpty();
+
+        createTable(phantomTable);
+        JdbcTableHandle tableHandle = jdbcClient.getTableHandle(SESSION, phantomTable).orElseThrow();
+        JdbcColumnHandle phantomColumn = addColumn(tableHandle);
+        assertThat(cachingJdbcClient.getColumns(SESSION, tableHandle))
+                .containsExactly(phantomColumn);
+
+        cachingJdbcClient.renameColumn(SESSION, tableHandle, phantomColumn, "new_column_name");
+
+        JdbcColumnHandle newColumnHandle = JdbcColumnHandle.builderFrom(phantomColumn)
+                .setColumnName("NEW_COLUMN_NAME")
+                .build();
+
+        assertThat(cachingJdbcClient.getColumns(SESSION, tableHandle))
+                .containsExactly(newColumnHandle);
+
+        cachingJdbcClient.dropTable(SESSION, tableHandle);
```

Comment:
`dropTable(phantomTable);`

Reply:
Won't work here. If I don't drop table via cachingJdbcClient then cache won't be invalidated